measures/UpgradeCosts/measure.rb

```diff
@@ -0,0 +1,282 @@
+      runner.registerInfo("Registering (blank) for #{upgrade_cost_name}.")
+      return true
+    end
+
+    # Obtain cost multiplier values and calculate upgrade costs
+    upgrade_cost = 0.0
+    option_cost_pairs.keys.each do |option_num|
+      option_cost = 0.0
+      option_cost_pairs[option_num].each do |cost_value, cost_mult_type|
+        cost_mult = get_cost_multiplier(cost_mult_type, hpxml, runner)
+        total_cost = cost_value * cost_mult
+        next if total_cost == 0
+
+        option_cost += total_cost
+        runner.registerInfo("Upgrade cost addition: $#{cost_value} x #{cost_mult} [#{cost_mult_type}] = #{total_cost}.")
+      end
+      upgrade_cost += option_cost
+
+      # Save option cost/lifetime to results.csv
+      next unless option_cost != 0
+      option_cost = option_cost.round(2)
+      option_cost_name = 'option_%02d_cost_usd' % option_num
+      register_value(runner, option_cost_name, option_cost)
+      runner.registerInfo("Registering #{option_cost} for #{option_cost_name}.")
+      next unless (not option_lifetimes[option_num].nil?) && (option_lifetimes[option_num] != 0)
+      lifetime = option_lifetimes[option_num].round(2)
+      option_lifetime_name = 'option_%02d_lifetime_yrs' % option_num
+      register_value(runner, option_lifetime_name, lifetime)
+      runner.registerInfo("Registering #{lifetime} for #{option_lifetime_name}.")
+    end
+    upgrade_cost = upgrade_cost.round(2)
+    register_value(runner, upgrade_cost_name, upgrade_cost)
+    runner.registerInfo("Registering #{upgrade_cost} for #{upgrade_cost_name}.")
+
+    return true
+  end
+
+  def get_cost_multiplier(cost_mult_type, hpxml, runner)
+    cost_mult = 0.0
+    if cost_mult_type == 'Fixed (1)'
+      cost_mult += 1.0
+    elsif cost_mult_type == 'Wall Area, Above-Grade, Conditioned (ft^2)'
+      hpxml.walls.each do |wall|
+        next unless wall.is_thermal_boundary
+
+        cost_mult += wall.area
+      end
+    elsif cost_mult_type == 'Wall Area, Above-Grade, Exterior (ft^2)'
+      hpxml.walls.each do |wall|
+        next unless wall.is_exterior
+
+        cost_mult += wall.area
+      end
+    elsif cost_mult_type == 'Wall Area, Below-Grade (ft^2)'
+      hpxml.foundation_walls.each do |foundation_wall|
+        cost_mult += foundation_wall.area
+      end
+    elsif cost_mult_type == 'Floor Area, Conditioned (ft^2)'
+      cost_mult += hpxml.building_construction.conditioned_floor_area
+    elsif cost_mult_type == 'Floor Area, Attic (ft^2)'
+      hpxml.frame_floors.each do |frame_floor|
+        next unless frame_floor.is_thermal_boundary
+        next unless frame_floor.is_interior
+
+        if frame_floor.is_floor
+          next unless [HPXML::LocationAtticVented, HPXML::LocationAtticUnvented].include?(frame_floor.interior_adjacent_to)
+        elsif frame_floor.is_ceiling
+          next unless [HPXML::LocationAtticVented, HPXML::LocationAtticUnvented].include?(frame_floor.exterior_adjacent_to)
+        end
```

Comment:
The attic floor is always considered a ceiling framefloor, so you should be able to simplify this as:

```
        next unless frame_floor.is_thermal_boundary
        next unless frame_floor.is_interior
        next unless frame_floor.is_ceiling
        next unless [HPXML::LocationAtticVented, HPXML::LocationAtticUnvented].include?(frame_floor.exterior_adjacent_to)
```